### ISSUE TITLE
QueryBuilder does not translate IDs of relational Objects into foreign_keys ([object Object] instead of object.id)

### ISSUE THREAD

**Issue by woltob** [mikro-orm/mikro-orm#1903]
**Describe the bug**

This code: 

```
... 

    let deviceLocation: DeviceLocation;
    const em = this.em.fork(true);
    const qb = em.createQueryBuilder(DeviceLocation);
    args.deviceLocationInputs.forEach((deviceLocationInput) => {
      deviceLocation = em.create(DeviceLocation, {
        timestampInMs: deviceLocationInput.timestampInMs,
        device: device, // this is not null or undefined and should produce device.id
      });
      qb.insert(deviceLocation).onConflict().ignore().execute();
    });
... 
```

produces the following error (the query is wrong, relational objects are not translated to foreign_key IDs:

```
insert into "device_location" 

("id", "timestamp_in_ms", "device_id") 

values 

( 'fecbfce7-d9e6-44d9-bd0d-eedcecd753ea', 1622715787837.117, '[object Object]') 

on conflict () do nothing returning "id"
```

Entity:

```

export type DeviceLocationDTO = {
  timestampInMs: number;
  device: Device;
};

@Entity()
@Unique({
  name: 'device_location_device_id_ts_in_ms_unique',
  properties: ['device', 'timestampInMs'],
})

// BaseEntity has the id (primary key)
export class DeviceLocation extends BaseEntity implements DeviceLocationDTO {

  @ManyToOne({
    entity: () => Device,
    cascade: [Cascade.PERSIST],
    strategy: LoadStrategy.JOINED,
    nullable: false,
  })
  public device: Device;

  .... 

  @Property({ type: Number, columnType: 'double precision', nullable: false })
  timestampInMs: number;

```

**Stack trace**

```
(node:11260) UnhandledPromiseRejectionWarning: error: insert into "device_location" ("id", "timestamp_in_ms", "device_id") values ( '4087e9c5-456f-4f42-8e5e-ae8a74dcfbfa', 1622719787837.2236, '[object Object]') on conflict ("timestamp_in_ms", "device_id") do nothing returning "id" - invalid input syntax for type uuid: "[object Object]"

``` 

**To Reproduce**
1. Create entity with composite Unique index for two columns (one foreign key, one BigInt).
2. Run QueryBuilder query:
```
result = await qb
        .insert(deviceLocation)
        .onConflict(['timestampInMS', 'deviceId'])
        .ignore()
        .execute();
```
3. The entity object deviceLocation has a foreign key to device. A Device object is passed to it. The qb query does not translate the object device inside deviceLocation to device.id, leading to the above postgres query, where instead of device.id it uses [object Object], meaning that it tries to print the device object as a string.

**Expected behavior**

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | v14.15.5 |
| typescript | ^4.2.3 |
| mikro-orm |  see below  |
| your-driver | postgres, see below |

"@mikro-orm/cli": "^4.5.5",
"@mikro-orm/core": "^4.5.5",
"@mikro-orm/nestjs": "^4.2.0",
"@mikro-orm/postgresql": "^4.5.5",
"@mikro-orm/reflection": "^4.5.5"

**Comment by B4nan** [comment]
This is unfortunately not enough to reproduce, please provide a failing test case as in https://github.com/mikro-orm/mikro-orm/tree/master/tests/issues.

The reproduction needs to be **minimal** (so remove everything unrelated, mainly no GQL), as well as **complete**.

**Comment by woltob** [comment]
I updated the issue above. It should have all the necessary information. Here in short again:

Two entities, DeviceLocation and Device exist, where DeviceLocation has a Foreign Key device_id to Device. new DeviceLocation({device: device, timestampInMs: 123123}) produces an entity object that can persisted. 

However, using query builder, the device object is not translated to device_id but rather to "[object Object]", likely meaning that device.id is not put in the query but rather the device object as a whole.

**Comment by woltob** [comment]
BUMP for another update. Super simplified now.

**Comment by B4nan** [comment]
Can you please retest with `4.5.6-dev.11`? I was fixing similar bug and the fix was in a helper method, so it might as well have fixed this issue too.

**Comment by B4nan** [comment]
Ping

**Comment by B4nan** [comment]
Closing due to inactivity, if you still see the same issue on latest version, please provide the test case I was asking for - again, it should be complete, e.g. entity definitions for everything you refer to, not like in your examples here, and minimal at the same time, so remove everything unrelated, but keep it executable.

But I believe it should be fixed already as it feels the same as #1902.
